src/irep2/irep2.h

```diff
@@ -177,60 +204,20 @@ template <class T>
 class irep_container : public std::shared_ptr<T>
 {
 public:
-  irep_container() : std::shared_ptr<T>()
-  {
-  }
-
-  template <class Y>
-  explicit irep_container(Y *p) : std::shared_ptr<T>(p)
-  {
-  }
+  constexpr irep_container() = default;
+  constexpr irep_container(const irep_container &ref) = default;
+  constexpr irep_container(irep_container &&ref) = default;
 
-  template <class Y>
-  explicit irep_container(const Y *p) : std::shared_ptr<T>(const_cast<Y *>(p))
-  {
-  }
+  irep_container &operator=(irep_container const &ref) = default;
+  irep_container &operator=(irep_container &&ref) = default;
```

Comment:
constexpr?

Reply:
We can mark it constexpr, but it won't help. It needs to copy/move the underyling std::shared_ptr, which [doesn't have](https://en.cppreference.com/w/cpp/memory/shared_ptr/operator%3D) a constexpr operator=().